### PR TITLE
Fix #4007 increase JPA extension test coverage

### LINKED ISSUE #4007: Increase JPA extension test coverage
- [ ] Refactor tests so it's not uses ProducerTemplate and ConsumerTemplate.
- [ ] JpaConsumer right now have 0% coverage
- [ ] Add test scenario to cover transactions
- [ ] Add test scenario to cover JpaMessageIdRepository
- [ ] Add test scenario to cover usage of **@Consumed and** **@PreConsumed** [annotations](https://camel.apache.org/components/3.18.x/jpa-component.html#_consuming_from_the_endpoint) on entity methods.
- [ ] Modify tests so that they can be run with any external database. This will be used for database certification as we do with [SQL extension](https://github.com/apache/camel-quarkus/tree/main/integration-tests/sql).

### PR DESCRIPTION
**Pull request by osmman**

### CODE REVIEW

File: integration-tests/jpa/src/test/java/org/apache/camel/quarkus/component/jpa/it/JpaTest.java [hunk 1]
```diff
@@ -16,63 +16,153 @@
...
+    @Test
+    public void testProducerNativeQuery() {
+        RestAssured.get("/jpa/fruit/native/2")
+                .then()
+                .statusCode(200)
+                .body("name", contains(FRUITS[1]));
+    }
+
+    @Test
+    public void testConsumer() {
```
Comment (aldettinger) [inline integration-tests/jpa/src/test/java/org/apache/camel/quarkus/component/jpa/it/JpaTest.java:84]: Does this one test `findById` ? If yes, maybe we could rename the test ?

**Comment by aldettinger** [comment]
Nice addition, most documented features seems to be tested with this PR :+1:

File: integration-tests/jpa/src/test/java/org/apache/camel/quarkus/component/jpa/it/JpaTest.java (hunk 1 above)
Comment (osmman) [reply integration-tests/jpa/src/test/java/org/apache/camel/quarkus/component/jpa/it/JpaTest.java:84]: It tests `from("jpa:Fruit?namedQuery=unprocessed)` route with usage of `@Consumed` and `@PreConsumed` annotations.

Comment (aldettinger) [reply integration-tests/jpa/src/test/java/org/apache/camel/quarkus/component/jpa/it/JpaTest.java:84]: Indeed, with the last comment, the intent is clearer now. Renaming the test to something like `testConsumedAnnotations()` could possibly make it even clearer.

**Review by aldettinger — APPROVED** [review]

**Review by zbendhiba — APPROVED** [review]

**Review by zhfeng — APPROVED** [review]

**Comment by zhfeng** [comment]
Thanks @osmman and it is great work!

**Pull request merged**